File: eval_rpn/eval_rpn.ts
export {};

function evalRpn(tokens: string[]) {
  const operand = new Set(["+", "-", "*", "/"]);
  const stack: number[] = [];
  tokens.forEach((token) => {
    if (operand.has(token)) {
        const a = stack.pop()!;
        const b = stack.pop()!;
      switch (token) {
        case "+":
          stack.push(b + a);
          break;
        case "-":
          stack.push(b - a);
          break;
        case "*":
          stack.push(b * a);
          break;
        case "/":
          stack.push(Math.trunc(b / a));
          break;
        default:
          break;
      }
    } else {
      stack.push(Number(token));
    }
  });

  console.log("Value of the expression =", stack.pop());
}
const token = ["10", "6", "9", "3", "+", "-11", "*", "/", "*", "17", "+", "5", "+"];
evalRpn(token);

File: eval_rpn/eval_rpn.py
def eval_rpn(tokens):
    operand = {"+", "-", "*", "/"}
    stack = []
    
    for token in tokens:
        if token in operand:
            a = stack.pop()
            b = stack.pop()
            if token == "+":
                stack.append(b + a)
            elif token == "-":
                stack.append(b -a)
            elif token == "*":
                stack.append(b * a)
            elif token == "/":
                stack.append(int(b / a))
        else:
            stack.append(int(token))
    print("Value of the expression =", stack.pop())
            
            
token = ["10", "6", "9", "3", "+", "-11", "*", "/", "*", "17", "+", "5", "+"]
eval_rpn(token)